pkg/meta/base.go

```diff
@@ -526,27 +647,116 @@ func (m *baseMeta) deleteSlice(chunkid uint64, size uint32) {
 	}
 }
 
-func (m *baseMeta) Readdir(ctx Context, inode Ino, plus uint8, entries *[]*Entry) syscall.Errno {
-	inode = m.checkRoot(inode)
-	var attr Attr
-	if err := m.GetAttr(ctx, inode, &attr); err != 0 {
-		return err
+func (m *baseMeta) toTrash(parent Ino) bool {
+	return m.fmt.TrashDays > 0 && !isTrash(parent)
```

Comment:
Where do we check that only root can delete a trash?

Reply:
Trash and SubTrash directory are created with uid/gid = 0 and mode = 0555

Reply:
8f2f1b2